### PR TITLE
SNOW-1736920 Fix bindings uploading and error handling for GCP

### PR DESCRIPTION
**Pull request by sfc-gh-dstempniak**
### Description
Please explain the changes you made here.

### Checklist
- [x] Code compiles correctly
- [x] Code is formatted according to [Coding Conventions](../blob/master/CodingConventions.md)
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing (`dotnet test`)
- [x] Extended the README / documentation, if necessary
- [x] Provide JIRA issue id (if possible) or GitHub issue id in PR name

### CODE REVIEW

File: Snowflake.Data/Core/FileTransfer/StorageClient/SFGCSClient.cs [hunk 1]
```diff
@@ -504,12 +504,16 @@ private SFFileMetadata HandleFileHeaderErrForGeneratedUrls(WebException ex, SFFi
         /// <returns>File Metadata</returns>
         private SFFileMetadata HandleUploadFileErr(WebException ex, SFFileMetadata fileMetadata)
         {
-            Logger.Error("Failed to upload file: " + ex.Message);
+            Logger.Error("Failed to upload file. ", ex);
```
Comment (sfc-gh-pbulawa) [inline Snowflake.Data/Core/FileTransfer/StorageClient/SFGCSClient.cs:507]: Are you sure that the stack trace does not contain any sensitive data, e.g. encryption material?

File: Snowflake.Data/Core/SFStatement.cs [hunk 2]
```diff
@@ -383,11 +383,14 @@ internal async Task<SFBaseResultSet> ExecuteAsync(int timeout, string sql, Dicti
                     SFBindUploader uploader = new SFBindUploader(SfSession, _requestId);
                     await uploader.UploadAsync(bindings, cancellationToken).ConfigureAwait(false);
                     _bindStage = uploader.getStagePath();
-                    ClearQueryRequestId();
                 }
                 catch (Exception e)
                 {
-                    logger.Warn("Exception encountered trying to upload binds to stage. Attaching binds in payload instead. {0}", e);
+                    logger.Warn($"Exception encountered trying to upload binds to stage. Attaching binds in payload instead.", e);
```
Comment (sfc-gh-pbulawa) [inline Snowflake.Data/Core/SFStatement.cs:389]: I think that maybe here we should also explicitly provide the `e.Message,` not the whole stack trace.

File: Snowflake.Data/Core/SFStatement.cs [hunk 3]
```diff
@@ -532,13 +535,14 @@ private SFBaseResultSet ExecuteSqlOtherThanPutGet(int timeout, string sql, Dicti
...
                         uploader.Upload(bindings);
                         _bindStage = uploader.getStagePath();
-                        ClearQueryRequestId();
                     }
                     catch (Exception e)
                     {
-                        logger.Warn(
-                            "Exception encountered trying to upload binds to stage. Attaching binds in payload instead. {0}",
-                            e);
+                        logger.Warn( $"Exception encountered trying to upload binds to stage. Attaching binds in payload instead.", e);
```
Comment (sfc-gh-pbulawa) [inline Snowflake.Data/Core/SFStatement.cs:541]: Same here

File: Snowflake.Data/Core/FileTransfer/StorageClient/SFGCSClient.cs (hunk 1 above)
Comment (sfc-gh-dstempniak) [reply Snowflake.Data/Core/FileTransfer/StorageClient/SFGCSClient.cs]: I'm not. I reverted this change.

File: Snowflake.Data/Core/SFStatement.cs (hunk 2 above)
Comment (sfc-gh-dstempniak) [reply Snowflake.Data/Core/SFStatement.cs]: done

File: Snowflake.Data/Core/SFStatement.cs (hunk 3 above)
Comment (sfc-gh-dstempniak) [reply Snowflake.Data/Core/SFStatement.cs]: done

**Review by sfc-gh-pbulawa — APPROVED** [review]

**Pull request merged**